incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -399,12 +399,22 @@ func (r *ConfigReconciler) setTypeStatuses(inst *api.HNCConfiguration) {
 			Mode:       ts.GetMode(), // may be different from the spec if it's implicit
 		}
 
-		// Only add counts if we're not ignoring this type
+		// Only add NumPropagatedObjects if we're not ignoring this type
 		if ts.GetMode() != api.Ignore {
 			numProp := ts.GetNumPropagatedObjects()
 			status.NumPropagatedObjects = &numProp
 		}
 
+		// Only add NumSourceObjects if we are propagating objects of this type.
+		if ts.GetMode() == api.Propagate {
+			numSrc := 0
+			nms := r.Forest.GetNamespaces()
+			for _, ns := range nms {
```

Comment:
Does it make sense to count the source objects in the reconciler at the same time we count the propagated objects instead? e.g. in addition to `recordPropagatedObject` and `removePropagatedObject`, we could have a `<record|remove>SourceObject` as well?

Reply:
Did you mean we call `<record|remove> SourceObject` next to `<record|remove>PropagatedObject>` in the places where `<record|remove>PropagatedObject>` are called? 

I don't think they should be counted at the same place. Looks like propagated objects and source objects are handled in different code paths.  For example, if we only have one namespace and we add a source object, if we only rely on `<record|remove>PropagatedObject>` to signal the config reconciler, the `NumSourceObjects` won't be updated because we will not go through the code path that invokes any of the `<record|remove>PropagatedObject>`.

Feel free to let me know if I miss anything or I misunderstand your comment.

Reply:
Yeah not in exactly the same places, I just mean the general idea of storing an up-to-date count in the object reconciler.

Reply:
Actually, it's pretty clear the way it is now, maybe best just to leave it.